Sources/TuistCache/ContentHashing/DependencyContentHasher.swift

```diff
@@ -19,31 +20,35 @@ public final class DependenciesContentHasher: DependenciesContentHashing {
 
     // MARK: - DependenciesContentHashing
 
-    public func hash(dependencies: [Dependency]) throws -> String {
-        let hashes = dependencies.map { try? hash(dependency: $0) }
+    public func hash(dependencies: [Dependency], sourceRootPath: AbsolutePath) throws -> String {
+        let hashes = dependencies.map { try? hash(dependency: $0, sourceRootPath: sourceRootPath) }
         return hashes.compactMap { $0 }.joined()
     }
 
     // MARK: - Private
 
-    private func hash(dependency: Dependency) throws -> String {
+    private func hash(dependency: Dependency, sourceRootPath: AbsolutePath) throws -> String {
```

Comment:
I think we need to rewrite the logic here. When we hash a framework we need to hash the file, not the path. A path doesn't tell us anything on whether the file framework/library has changed or not. Hence, we don't need to go from absolute paths to relative paths, the problem is that the implementation is not right.

Reply:
While I agree we should hash the framework and not the path, this is still an improvement, it makes the current hashing implementation consistent on multiple machines. I'd proceed, merge this PR and open an issue to hash the framework instead.

Reply:
I disagree. We are building upon something that is fundamentally incorrect. Hashing is a critical piece of caching and it needs to be foundationally well designed. Merging this PR is disregarding that and potentially leading to more issues.